packages/styled-components/src/types.ts

```diff
@@ -36,6 +36,11 @@ export interface StyledOptions<R extends Runtime, Props extends object> {
 
 export type Dict<T> = { [key: string]: T };
 
+// Allows for jsx ignored attributes such as data-* and aria-*
+export interface HyphenatedAttributes {
+  [key: `${string}-${string}`]: unknown;
```

Comment:
Is there a way to limit this more to only known aria attributes? `data-${string}` can stay since that's an open field

Reply:
It looks like you could pull in `React.AriaAttributes`

Reply:
Done